pkg/whip/whip_handler.go

```diff
@@ -514,14 +510,20 @@ func (h *whipHandler) runSession(ctx context.Context) error {
 
 	result := make(chan error, 1)
 
-	for _, th := range h.trackHandlers {
+	h.trackLock.Lock()
+	for td, th := range h.trackHandlers {
 		err := th.Start(func(err error) {
```

Comment:
just double checking that the closure runs async..

Reply:
Yes, `Start` starts a coroutine.